dace/codegen/targets/xilinx.py

```diff
@@ -194,6 +196,7 @@ def get_generated_codeobjects(self):
                            XilinxCodeGen,
                            "Xilinx",
                            target_type="device"))
+            bank_assignment_code.clear()
```

Comment:
why is the `clear()` here necessary?

Reply:
Because we are creating the bankassignments for each kernel seperately. If clear is missing here, then we will add all assignments from the previous kernels to the kernels generated later (applies only for SDFGs with multiple states running on the SDFG

Reply:
*on the FPGA